nixos/doc/manual/release-notes/rl-2311.section.md

```diff
@@ -699,46 +1330,104 @@ The module update takes care of the new config syntax and the data itself (user
   [`cmakeFeature`](https://nixos.org/manual/nixpkgs/unstable#function-library-lib.strings.cmakeFeature).
-- [`lib.trivial`](https://nixos.org/manual/nixpkgs/unstable#sec-functions-library-trivial): New function:
+- [`lib.trivial`](https://nixos.org/manual/nixpkgs/unstable#sec-functions-library-trivial):
+  New function:
   [`mirrorFunctionArgs`](https://nixos.org/manual/nixpkgs/unstable#function-library-lib.trivial.mirrorFunctionArgs).
 - `lib.systems`: New function:
   [`equals`](https://nixos.org/manual/nixpkgs/unstable#function-library-lib.systems.equals).
-- [`lib.options`](https://nixos.org/manual/nixpkgs/unstable#sec-functions-library-options): Improved documentation for
+- [`lib.options`](https://nixos.org/manual/nixpkgs/unstable#sec-functions-library-options):
+  Improved documentation for
   [`mkPackageOption`](https://nixos.org/manual/nixpkgs/unstable#function-library-lib.options.mkPackageOption).
 
   [`mkPackageOption`](https://nixos.org/manual/nixpkgs/unstable#function-library-lib.options.mkPackageOption).
   now also supports the `pkgsText` attribute.
 
 Module system:
-- Options in the `options` module argument now have the `declarationPositions` attribute
-  containing the position where the option was declared:
-  ```
-  $ nix repl -f '<nixpkgs/nixos>'
-  [...]
-  nix-repl> :p options.environment.systemPackages.declarationPositions
-  [ {
-    column = 7;
-    file = "/nix/store/vm9zf9wvfd628cchj0hdij1g4hzjrcz9-source/nixos/modules/config/system-path.nix";
-    line = 62;
-  } ]
-  ```
-  Not to be confused with `definitionsWithLocations`, which is the same but for option _definitions_.
+- Options in the `options` module argument now have the `declarationPositions`
+  attribute containing the position where the option was declared: ``` $ nix
+  repl -f '<nixpkgs/nixos>' [...] nix-repl> :p
+  options.environment.systemPackages.declarationPositions [ { column = 7; file =
+  "/nix/store/vm9zf9wvfd628cchj0hdij1g4hzjrcz9-source/nixos/modules/config/system-path.nix";
+  line = 62; } ] ``` Not to be confused with `definitionsWithLocations`, which
+  is the same but for option _definitions_.
 - Improved error message for option declarations missing `mkOption`
 
-### Deprecations {#sec-release-23.11-lib-deprecations}
+#### Deprecations {#sec-release-23.11-lib-deprecations}
+
+- `lib.meta.getExe pkg` (also available as `lib.getExe`) now gives a warning if
+  `pkg.meta.mainProgram` is not set, but it continues to default to the
+  derivation name. Nixpkgs accepts PRs that set `meta.mainProgram` on packages
+  where it makes sense. Use `lib.getExe' pkg "some-command"` to avoid the
+  warning and/or select a different executable.
+
+### Nixpkgs Internals {#sec-release-23.11-nixpkgs-internals}
+
+- The use of `sourceRoot = "source";`, `sourceRoot = "source/subdir";`, and
+  similar lines in package derivations using the default `unpackPhase` is
+  deprecated as it requires `unpackPhase` to always produce a directory named
```

Comment:
As mention before, all of these items aren't internal, I don't think it makes sense to have this section.

Reply:
Could you propose a better fitting title?

Reply:
I didn't move those because I wasn't too sure where to put them. I think the reason they have been marked "internal" it's because they refer to package or service making, instead of the packages and options themselves. Where would they fit?

Reply:
I'll make a PR!

Reply:
After all the cleanup, I see them either as breaking changes or other notable changes, so I'll move them and update outline and the TOC. I think we can still do better on how to better segment the release notes, but I'll put those efforts going into the future into the autogeneration and then the discussion can move into proper tagging of the documentation of changes.